PharmacyApi/Program.cs

```diff
@@ -32,12 +28,11 @@
     options.UseSqlServer(connectionString);
 });
 
-builder.Services.AddTransient<IPharmacyService, PharmacyService>();
+builder.Services.AddScoped<IPharmacyService, PharmacyService>();
+builder.Services.AddScoped<IPharmacyDbContext, PharmacyDbContext>();
```

Comment:
Scoped - you will have problems when more than 4 people hit your api at the same time...
It takes a while to new-up a DbContext... then again is your context structure going to change at runtime?... nope...